src/main/groovy/util/Node.java

```diff
@@ -537,26 +539,40 @@ private NodeList getByName(String name) {
 
     /**
      * Provides a collection of all the nodes in the tree
-     * using a depth first traversal.
+     * using a depth-first preorder traversal.
      *
      * @return the list of (depth-first) ordered nodes
      */
     public List depthFirst() {
+        return depthFirst(true);
+    }
+
+    /**
+     * Provides a collection of all the nodes in the tree
+     * using a depth-first traversal.
+     *
+     * @param preorder if false, a postorder depth-first traversal will be performed
```

Comment:
You'd traverse the tree (traversal type doesn't matter), adding each node to a TreeSet, then return a List of the TreeSet.  Kinda funky, I'm not sure it's a good idea.  It's just the only way I could think of to offer in-order traversal for non-binary trees.